### PR TITLE
Add status to situation api

### PR DESCRIPTION
**Pull request by BenjaminJ**
o add status to Situation.java
o moved Enum Status to datasource/api
o add situationStatus to alarm event (ApiMapper.java)
o retrieve situationStatus from alarm event (ApiMapper.java)
o Default situationStatus is CREATED

ALEC-135

### CODE REVIEW

File: datasource/common/src/main/java/org/opennms/alec/datasource/common/ImmutableSituation.java [hunk 1]
```diff
@@ -229,6 +240,11 @@ public String getDiagnosticText() {
         return diagnosticText;
     }
 
+    @Override
+    public Status getStatus() {
+        return Optional.ofNullable(status).orElse(Status.CREATED);
+    }
+
     @Override
     public boolean equals(Object o) {
```
Comment (aramosvizcarra) [inline datasource/common/src/main/java/org/opennms/alec/datasource/common/ImmutableSituation.java:249]: You need to add status to equals as well :)

**Review by aramosvizcarra — CHANGES_REQUESTED** [review]
Looks good to me! just one minor change.

File: datasource/common/src/main/java/org/opennms/alec/datasource/common/ImmutableSituation.java (hunk 1 above)
Comment (BenjaminJ) [reply datasource/common/src/main/java/org/opennms/alec/datasource/common/ImmutableSituation.java:249]: Good catch don't how I missed that, thank you

**Review by aramosvizcarra — APPROVED** [review]
LGTM!

**Review by amay94 — APPROVED** [review]
LGTM

**Pull request merged**